src/message/message.cc

```diff
@@ -50,9 +52,22 @@ void CRMessageBase::Unsubscribe(const std::string& message_type, CRNodeBase* nod
     subscription_list.pop_back();
 }
 
-const CRMessageBase::subscription_list_t* CRMessageBase::GetSubscriptionList(const std::string& message_type) {
-    auto subscription_find = impl::subscription_map.find(message_type);
-    if (subscription_find == impl::subscription_map.end()) [[unlikely]] {
+cr_timestamp_nsec_t CRMessageBase::GetLatestDeliveredTimeImpl(const std::string& message_type) {
+    constexpr cr_timestamp_nsec_t kDefaultDeliveredTime = 0;
+
+    auto* subscription_map        = GetSubscriptionMap();
+    auto  subscription_map_search = subscription_map->find(message_type);
+    if (subscription_map_search == subscription_map->end()) {
+        LOG(WARNING) << __func__ << ": message '" << message_type << "' is unknown.";
+        return kDefaultDeliveredTime;
+    }
+    return subscription_map_search->second.latest_delivered_time_.load();
+}
+
+CRMessageBase::SubscriptionInfo* CRMessageBase::GetSubscriptionInfo(const std::string& message_type) {
+    auto* subscription_map  = GetSubscriptionMap();
+    auto  subscription_find = subscription_map->find(message_type);
+    if (subscription_find == subscription_map->end()) [[unlikely]] {
```

Comment:
This hint is not accurate. What if the user intentionally send a lot of wrong queries, or even use it to check the existence of a key? Don't hint unless it's proven to be unlikely.